tests/integration/test_logcollector/test_configuration/test_basic_configuration_reconnect_time.py

```diff
@@ -0,0 +1,142 @@
+from wazuh_testing.utils import callbacks, configuration
+from wazuh_testing.utils.services import control_service
+from wazuh_testing.utils.file import truncate_file
+
+from . import TEST_CASES_PATH, CONFIGURATIONS_PATH
+
+# Marks
+if sys.platform == WINDOWS:
+    pytestmark = pytest.mark.tier(level=0)
+else:
+    pytestmark = [pytest.mark.skip, pytest.mark.tier(level=0)]
+
+# Test metadata, configuration and ids.
+cases_path = Path(TEST_CASES_PATH, 'cases_basic_configuration_reconnect_time.yaml')
+config_path = Path(CONFIGURATIONS_PATH, 'wazuh_basic_configuration_reconnect_time.yaml')
+test_configuration, test_metadata, test_cases_ids = configuration.get_test_cases_data(cases_path)
+test_configuration = configuration.load_configuration_template(config_path, test_configuration, test_metadata)
+
+problematic_values = ['44sTesting', '9hTesting', '400mTesting', '3992']
+
+# Test daemons to restart.
+daemons_handler_configuration = {'all_daemons': True}
+
+# Test function.
+@pytest.mark.parametrize('test_configuration, test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_configuration_reconnect_time(test_configuration, test_metadata, truncate_monitored_files,
+                                      set_wazuh_configuration, daemons_handler_module):
+    '''
+    description: Check if the 'wazuh-logcollector' daemon detects invalid settings for the 'reconnect_time' tag.
+                 For this purpose, the test will set a 'localfile' section using both valid and invalid values
+                 for that tag. Finally, the test will verify that the 'analyzing' event is triggered when using
+                 a valid value or if the 'invalid' event is generated when using an invalid one.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - get_configuration:
+            type: fixture
+            brief: Get configurations from the module.
+        - configure_environment:
+            type: fixture
+            brief: Configure a custom environment for testing.
+
+    assertions:
+        - Verify that the logcollector generates 'invalid' events when using invalid values
+          for the 'reconnect_time' tag.
+        - Verify that the logcollector monitors a log file when using valid values for the 'reconnect_time' tag.
+
+    input_description: A configuration template (test_basic_configuration_reconnect_time) is contained in an
+                       external YAML file (wazuh_basic_configuration.yaml). That template is combined with
+                       different test cases defined in the module. Those include configuration settings
+                       for the 'wazuh-logcollector' daemon.
+
+    expected_output:
+        - r'Analyzing event log.*'
+        - r'Invalid reconnection time value. Changed to .* seconds.'
+
+    tags:
+        - invalid_settings
+        - logs
+    '''
+
+    control_service('stop', daemon=LOGCOLLECTOR_DAEMON)
```

Comment:
As you do this in many tests (stop, truncate), can you create a fixture for it?

Reply:
Done